src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs

```diff
@@ -450,6 +490,17 @@ static bool TryGetInnermostReceiver(ExpressionSyntax node, ArrayBuilder<Identifi
                 rightNames.Clip(rightIndex + 1);
                 return lastName;
             }
+
+            static bool IsNullCheck(SyntaxNode? exp)
+            {
+                if (exp is BinaryExpressionSyntax bin && bin.OperatorToken.Kind() == ExclamationEqualsToken
```

Comment:
don't abbr.  can say `binaryExpression`.

Reply:
can check the kind directly.  like `is BinaryExpressionSyntax(SyntaxKind.NotEqualsExpression)`

Reply:
I tried this, it was breaking and I couldn't even get this to work in my IDE's evaluator during debugging, is this the correct why to do it?

Reply:
@CyrusNajmabadi hey, I am still waiting a response for this: are you sure this is the correct expression? I wasn't able to make it work.

Reply:
Updated.